templates/flavors/azure-cni-v1/patches/azure-machine-template-controlplane.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+  namespace: default
+spec:
+  template:
+    spec:
+      networkInterfaces:
+      - subnetName: cp-subnet-1
+        privateIPConfigs: 10 # extra IPs allocated so that core-dns pods, if scheduled, do not fail readiness probe
```

Comment:
is 10 a significant number here?

Reply:
Extra IPs were more of a buffer for my testing. 
I have updated the number to `4`. 
`4` because 
- first IP gets assigned to the node
- three other IPs can be used by the core-dns pods if all of them get scheduled on the control-plane node.

Reply:
What if other pods get scheduled to the control plane nodes later (either within the context of a new e2e test spec or outside e2e tests)? it might make sense to also set this to the value of maxPods for the node to be safe

Reply:
Thats a fair thought. Updated to `110`.